voila/handler.py

```diff
@@ -127,7 +140,9 @@ async def get_generator(self, path=None):
                 kernel_spec_manager=self.kernel_spec_manager,
             )
 
-            await gen.initialize(template=template_arg, theme=theme_arg)
+            done = await gen.initialize(template=template_arg, theme=theme_arg)
+            if not done:
+                raise tornado.web.HTTPError(404, 'NotebookRenderer has failed to initialize!')
```

Comment:
Instead of doing this, it might be nicer that `initialize` fails properly. I'm thinking [this method](https://github.com/voila-dashboards/voila/blob/63e0f91cde88a62af9130b2008b06ee01a3e12a5/voila/notebook_renderer.py#L298) could throw a proper error if it fails:

```python
    async def load_notebook(self, path):

        model = await ensure_async(self.contents_manager.get(path=path))
        if 'content' not in model:
            raise tornado.web.HTTPError(404, 'file not found')
        __, extension = os.path.splitext(model.get('path', ''))
        if model.get('type') == 'notebook':
            notebook = model['content']
            notebook = await self.fix_notebook(notebook)
            return notebook
        elif extension in self.voila_configuration.extension_language_mapping:
            language = self.voila_configuration.extension_language_mapping[extension]
            notebook = await self.create_notebook(model, language=language)
            return notebook
        
         ## THROW HERE
         else:
            raise tornado.web.HTTPError(404, 'NotebookRenderer has failed to initialize')
```

?

Also I'm wondering if a `404` is the right error? It feels like a file was found, but we failed to process it, should it be a `500`?

Reply:
Updated

Reply:
Thanks!